src/bytes.rs

```diff
@@ -58,27 +59,42 @@ const _: () = {
 
 extern "C" fn noop(_: *const (), _: usize) {}
 
+const IS_LITTLE_ENDIAN: bool = unsafe { mem::transmute::<u16, [bool; 2]>(1)[0] };
+
 impl<'a> Bytes<'a> {
     /// Constructs an empty slice.
     ///
     /// Since [`Bytes`] are immutable slices, this mostly serves to create default values.
     pub const fn empty() -> Self {
         Self::from_static([].as_slice())
     }
-    const fn is_interned(&self) -> bool {
-        (unsafe {
+
+    /// The maximum size of a slice that can be inlined in [`Bytes`].
+    ///
+    /// Most CPUs are little endian, so on 64bit systems, this will be `32`, and `16` on 32bit systems.
+    pub const MAX_INLINE_SIZE: usize = mem::size_of::<Bytes<'static>>()
+        - if IS_LITTLE_ENDIAN {
+            mem::size_of::<usize>()
+        } else {
+            1
+        };
+    const fn read_inline_length_byte(&self) -> u8 {
+        unsafe {
             (self as *const Self)
                 .cast::<u8>()
-                .add(core::mem::size_of::<Self>() - 1)
+                .add(Self::MAX_INLINE_SIZE)
                 .read()
-        } & 1)
-            != 0
+        }
     }
-    fn vtable(&self) -> Option<&'a BytesVt> {
-        if self.is_interned() {
+    const fn is_inlined(&self) -> bool {
+        (self.read_inline_length_byte() & 1) != 0
+    }
+    const fn vtable(&self) -> Option<&'a BytesVt> {
+        if self.is_inlined() {
             None
         } else {
-            unsafe { Some(self.vtable.as_ref()) }
+            // SAFETY: If the value is not
```

Comment:
Author **died** of a heart attack, but not before duly `git commit`ting and `git push`ing 😝

Reply:
drat, the one spot I let the apply suggestion run xD